src/commands/toolchain.rs

```diff
@@ -64,12 +86,27 @@ fn name_allowed(s: &str) -> Result<String> {
     }
 }
 
+fn channel_allowed(s: &str) -> Result<String> {
+    if s.is_empty() {
+        return Ok(s.to_string());
+    }
+    if toolchain_override::Channel::from_str(s).is_err() {
+        bail!(
+            "Invalid channel '{}', expected one of {}.",
+            s,
+            VALID_CHANNEL_STR,
+        );
+    }
+    Ok(s.to_string())
+}
+
 pub fn exec(command: ToolchainCommand) -> Result<()> {
     match command {
         ToolchainCommand::Install(command) => install(command)?,
         ToolchainCommand::New(command) => new(command)?,
         ToolchainCommand::Uninstall(command) => uninstall(command)?,
         ToolchainCommand::ListRevisions(command) => list_revisions(command)?,
+        ToolchainCommand::Export(command) => export(command, stdin().lock())?,
```

Comment:
What's the reason for locking the `stdin` here?

Reply:
I have updated to `BufReader::new(stdin())`